programs/create-genesis-ee/golos_operations.hpp

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+#include "golos_types.hpp"
+
+namespace cyberway { namespace golos {
+
+struct comment_operation {
+    account_name_type parent_author;
+    string parent_permlink;
+
+    account_name_type author;
+    string permlink;
+
+    string title;
+    string body;
+    string json_metadata;
+};
+
+struct custom_json_operation {
+    flat_set<account_name_type> required_auths;
+    flat_set<account_name_type> required_posting_auths;
+    string id; ///< must be less than 32 characters long
+    string json; ///< must be proper utf8 / JSON string.
+};
+
+// Currently not used, dump stores hash only
+// struct delete_comment_operation {
+//     account_name_type author;
+//     string permlink;
+// };
+
+struct transfer_operation {
+    account_name_type from;
+    /// Account to transfer asset to
+    account_name_type to;
+    /// The amount of asset to transfer from @ref from to @ref to
+    asset amount;
+
+    /// The memo is plain-text, any encryption on the memo is up to
+    /// a higher level protocol.
+    string memo;
+};
+
+struct vote_operation {
+    account_name_type voter;
+    account_name_type author;
+    string permlink;
+    int16_t weight = 0;
+};
+
+struct author_reward_operation {
+    account_name_type author;
+    string permlink;
+    asset sbd_payout;
+    asset steem_payout;
+    asset vesting_payout;
+};
+
+struct comment_benefactor_reward_operation {
+    account_name_type benefactor;
+    account_name_type author;
+    string permlink;
+    asset reward;
+};
+
+struct curation_reward_operation {
+    account_name_type curator;
+    asset reward;
+    account_name_type comment_author;
+    string comment_permlink;
+};
+
+struct auction_window_reward_operation {
+    asset reward;
+    account_name_type comment_author;
+    string comment_permlink;
+};
+
+struct total_comment_reward_operation {
+    account_name_type author;
+    string permlink;
+    asset author_reward;
+    asset benefactor_reward;
+    asset curator_reward;
```

Comment:
If I understand correctly, you don't know that fc::raw_pack/raw_unpack doesn't have any start-stop markers for structures.
You can simplify your code, and add net_rshares here.

Reply:
@afalaleev I was viewing how it looks in binary, but I was not thought about such solution. You proposing to add field to this operation but not to Golos's operation. IMHO such "asymmetric" solution is strange. golos_operations.hpp can be reused in somewhere other, let's make it correct.

Reply:
It is clean for developers who knows how it works.

Reply:
You can add comment about absent markers on start-stop structure.

Reply:
@afalaleev I will add net rshares to real Golos operation.